docs/nix.md

```diff
@@ -0,0 +1,56 @@
+# Overview
+
+`nix` ecosystem replaces fully or partially `cargo install`, `rustup`, `cargo make`, `docker build`, `sh`, `rust-cache`, `sccache`, `github actions`, `github workflow runs`, `crates.io`, `npmjs.com` etc.
+
+If to target `nixos` then `ansible`, `ssh`, `terraform`.
```

Comment:
What does this mean?

Reply:
all this was used in our CI/CD. what i tell is that if you want to use any of above, please consider using nix as its provides all of this by design. 

so people can undrestand what to find in nix by analogy.